### ISSUE TITLE
Unable to extract from signature when long keys are used in ~/.gnupg/gpg.conf

### ISSUE THREAD

**Issue by dysinger** [commercialhaskell/stack#1952]
If you are using the keyid-form 0xlong the text output format changes for gpg signing.

~~~
user@computer:~/src/github.com/commercialhaskell/stack% grep -i keyid-format ~/.gnupg/gpg.conf
# keyid-format 0xlong
user@computer:~/src/github.com/commercialhaskell/stack% gpg2 --output - --detach-sig --armor /home/user/src/github.com/commercialhaskell/stack/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/stack-1.0.4.2.tar.gz | gpg --verify - /home/user/src/github.com/commercialhaskell/stack/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/stack-1.0.4.2.tar.gz
gpg: Signature made Fri 25 Mar 2016 01:33:35 PM PDT using RSA key ID C156F68B
gpg: Good signature from "Tim Dysinger <tim@dysinger.net>" [ultimate]
gpg:                 aka "[jpeg image of size 4356]" [ultimate]
gpg:                 aka "Tim Dysinger <dysinger@gmail.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@positiveinertia.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@knewton.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@fpcomplete.com>" [ultimate]
Primary key fingerprint: DDB9 DB7D 11B7 5D1E 8DA9  959E 11A9 6A3C 6622 290C
     Subkey fingerprint: 30C2 1B41 62E8 AB7A 4F0C  F6DF 155E 7413 C156 F68B
user@computer:~/src/github.com/commercialhaskell/stack% grep -i keyid-format ~/.gnupg/gpg.conf
keyid-format 0xlong
user@computer:~/src/github.com/commercialhaskell/stack% gpg2 --output - --detach-sig --armor /home/user/src/github.com/commercialhaskell/stack/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/stack-1.0.4.2.tar.gz | gpg --verify - /home/user/src/github.com/commercialhaskell/stack/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/stack-1.0.4.2.tar.gz
gpg: Signature made Fri 25 Mar 2016 01:33:49 PM PDT
gpg:                using RSA key 0x155E7413C156F68B
gpg: Good signature from "Tim Dysinger <tim@dysinger.net>" [ultimate]
gpg:                 aka "[jpeg image of size 4356]" [ultimate]
gpg:                 aka "Tim Dysinger <dysinger@gmail.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@positiveinertia.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@knewton.com>" [ultimate]
gpg:                 aka "Tim Dysinger <tim@fpcomplete.com>" [ultimate]
Primary key fingerprint: DDB9 DB7D 11B7 5D1E 8DA9  959E 11A9 6A3C 6622 290C
     Subkey fingerprint: 30C2 1B41 62E8 AB7A 4F0C  F6DF 155E 7413 C156 F68B
user@computer:~/src/github.com/commercialhaskell/stack% 
~~~

**Comment by dysinger** [comment]
The GPG key-id long form doesn't work with stack. We should switch to using full fingerprints all the time.

**Comment by dysinger** [comment]
Corrected for < 7.10 and misc cleanup

**Comment by dysinger** [comment]
Added P1 because it's broken right now
